ui/components/app/incoming-trasaction-toggle/hooks.js

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from 'react';
+import { TEST_CHAINS } from '../../../../shared/constants/network';
+
+export const useIncomingTransactionToggle = ({
+  allNetworks,
+  incomingTransactionsPreferences,
+  setIncomingTransactionsPreferences,
+}) => {
+  const [networkPreferences, setNetworkPreferences] = useState(
+    generateIncomingNetworkPreferences(
+      incomingTransactionsPreferences,
+      allNetworks,
+    ),
+  );
```

Comment:
>Are we showing a toggle for every network they have added
Yes, indeed. 

>The incoming transactions controller only supports the networks in the ETHERSCAN_SUPPORTED_NETWORKS constant
````
    const { incomingTransactionsPreferences } =
      this.preferencesController.store.getState();
   const hasIncomingTransactionsFeatureEnabled = Boolean(
      incomingTransactionsPreferences[chainId],
    );
```
i believe this would prevent the issue, wdyt?